circe-akka-serializer/src/main/scala/org/virtuslab/ash/CirceAkkaSerializer.scala

```diff
@@ -0,0 +1,78 @@
+import io.circe.jawn.JawnParser
+import io.circe.{Decoder, Encoder, Printer}
+import org.reflections8.Reflections
+
+import java.io.NotSerializableException
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.NoSuchElementException
+import scala.reflect.{ClassTag, classTag}
+import scala.reflect.runtime.{universe => ru}
+
+abstract class CirceAkkaSerializer[Ser: ClassTag](actorSystem: ExtendedActorSystem)
+    extends SerializerWithStringManifest {
+  override val identifier = 42352
+  protected val logger = Logging(actorSystem, this.getClass)
+
+  val codecs: Seq[(ru.TypeTag[_ <: Ser], (Encoder[_ <: Ser], Decoder[_ <: Ser]))]
+  val manifestMigrations: Seq[(String, String)]
+
+  private lazy val parents = codecs.flatMap { x =>
+    val clazz = x._1.tpe.typeSymbol.asClass
+    val mirror = ru.runtimeMirror(getClass.getClassLoader)
+    val rootClazzName = mirror.runtimeClass(clazz).getName
+    def getAllSubclasses(clazz: ru.ClassSymbol): List[ru.ClassSymbol] = {
+      if (!clazz.isSealed)
+        List(clazz)
+      else
+        clazz :: clazz.knownDirectSubclasses.toList.flatMap(x => getAllSubclasses(x.asClass))
+    }
+    getAllSubclasses(clazz).map(x => (mirror.runtimeClass(x).getName, rootClazzName))
+  }.toMap
+  private lazy val codecsMap = codecs.map(x => x.copy(_1 = x._1.tpe.typeSymbol.fullName)).toMap
+  private lazy val manifestMap = manifestMigrations.toMap
+
+  private val parser = new JawnParser
+  private val printer = Printer.noSpaces
+
+  override def manifest(o: AnyRef): String = parents(o.getClass.getName)
+
+  override def toBinary(o: AnyRef): Array[Byte] = {
+    val (encoder, _) = getCodec(manifest(o))
+    val buffer = printer.printToByteBuffer(encoder.asInstanceOf[Encoder[AnyRef]](o), UTF_8)
+    val res = new Array[Byte](buffer.remaining)
+    buffer.get(res)
+    res
+  }
+
+  override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = {
+    val (_, decoder) = getCodec(manifestMap.getOrElse(manifest, manifest))
+    parser.parseByteArray(bytes).flatMap(_.as(decoder)).fold(e => throw e, identity).asInstanceOf[AnyRef]
+  }
+
+  private def getCodec(manifest: String): (Encoder[_], Decoder[_]) = {
+    codecsMap.get(manifest) match {
+      case Some(codec) => codec
+      case None        => throw new NotSerializableException("Tried to serialize/deserialize unrecognized serializable")
+    }
+  }
+
+  def scanForSerializables(packagePrefix: String): Unit = {
+    val reflections = new Reflections(packagePrefix)
+    val foundSubTypes = reflections.getSubTypesOf(classTag[Ser].runtimeClass)
+    val foundSerializables = CollectionConverter.convert(foundSubTypes).filterNot(_.isInterface)
+    foundSerializables.foreach { clazz =>
+      try {
+        getCodec(parents(clazz.getName))
+      } catch {
+        case _ @(_: NoSuchElementException | _: NotSerializableException) =>
+          logger.warning(s"No associated Register[A] found for ${clazz.getName} serializable")
```

Comment:
Same here, I wouldn't use the word `serializable` here, it'd be pretty confusing... just `class`

Reply:
Plus a hint on what to do